src/battle_script_commands.c

```diff
@@ -9776,7 +9761,10 @@ static void Cmd_various(void)
     case VARIOUS_SET_TERRAIN:
     {
         VARIOUS_ARGS(const u8 *failInstr);
-        HandleTerrainMove(gCurrentMove);
+        if (HandleTerrainMove(gBattleMoves[gCurrentMove].argument))
+            gBattlescriptCurrInstr += 7;
+        else
+            gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 3);
```

Comment:
Now that the script control flow has moved into this function these should be `gBattlescriptCurrInstr = cmd->next;` and `gBattlescriptCurrInstr = cmd->failInstr;`.

Reply:
Oops, thanks!